src/bindings/lib/Tree_sitter_output.ml

```diff
@@ -48,7 +48,9 @@ let of_ts_tree ts_tree =
     incr counter;
     !counter
   in
-  of_ts_node get_node_id root
+  let res = of_ts_node get_node_id root in
```

Comment:
this function needs also a comment in the .mli saying it's internally calling Tree.delete and so you should not
use anymore function on the deleted tree.

Reply:
Ideally we have an interface where we don't even need to see the `ts_tree`? For me that would be even better than using the trick from the OCaml Z3 bindings.